internal/rbd/controllerserver.go

```diff
@@ -922,10 +922,10 @@ func (cs *ControllerServer) DeleteVolume(
 	}
 
 	rbdVol, err := GenVolFromVolID(ctx, volumeID, cr, req.GetSecrets())
-	defer rbdVol.Destroy()
 	if err != nil {
 		return cs.checkErrAndUndoReserve(ctx, err, volumeID, rbdVol, cr)
 	}
+	defer rbdVol.Destroy()
```

Comment:
If there is an error, `GenVolFromVolID()` should not return a connected `rbdVol`... If that is the case, `GenVolFromVolID()` should be corrected too.

Reply:
> Please note that this case lead to the memory leak problem as rbdVol is returned even in case of error from GenVolFromVolID function

Oh I see, maybe GenVolFromVolID funtion returns error though, the connection has established.